src.common/Enums/Direction8.ts

```diff
@@ -9,8 +8,7 @@ export interface Direction8
 	readonly opposite: Direction8;
 }
 
-class Direction8Impl implements Direction8
-{
+class Direction8Impl implements Direction8 {
```

Comment:
Just a style thing, but calling the implemented class Direction8Impl feels wierd to me. I think I'd be happier with interfaces getting the different name e.g. IDirection8 and calling the implemented class Direction8. But it doesn't really matter either way, feel free to ignore this comment.

Reply:
(Also my comment is essentially unrelated to your actual change so maybe something to pick up in a future piece of work)

Reply:
I think I have an idea how to rewrite it to not use interface. I try to avoid prefixing interfaces with `I` nowadays because it doesn't bring any useful information (we already have `Entity` that is interface btw), here's a good read: https://stackoverflow.com/questions/31876947/confused-about-the-interface-and-class-coding-guidelines-for-typescript/41967120#41967120

Interestingly TypeScript's coding guidelines (as in the guidelines for contributing to TS itself) are against that too: https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#names